libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,235 @@
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Conditions;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Integration.ApplicationInsights.Core;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.Integration.AspNet.Core.Skills;
+using Microsoft.Bot.Builder.Integration.Runtime.Plugins;
+using Microsoft.Bot.Builder.Integration.Runtime.Settings;
+using Microsoft.Bot.Builder.Integration.Runtime.Skills;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Bot.Builder.Integration.Runtime.Extensions
+{
+    /// <summary>
+    /// Defines extension methods for <see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds bot runtime-related services to the application's service collection.
+        /// </summary>
+        /// <param name="services">The application's collection of registered services.</param>
+        /// <param name="configuration">The application configuration.</param>
+        public static void AddBotRuntime(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            // Component registrations must be added before the resource explorer is instantiated to ensure
+            // that all types are correctly registered. Any types that are registered after the resource explorer
+            // is instantiated will not be picked up otherwise.
+            ComponentRegistrations.Add();
+
+            // Configuration
+            string applicationRoot = configuration.GetSection(ConfigurationConstants.ApplicationRootKey).Value;
+            string defaultLocale = configuration.GetSection(ConfigurationConstants.DefaultLocale).Value;
+            string rootDialog = configuration.GetSection(ConfigurationConstants.RootDialogKey).Value;
+            
+            // Runtime settings. If no config is provided, we create basic runtime config with defaults.
+            var runtimeSettings = configuration.GetSection(ConfigurationConstants.RuntimeSettingsKey).Get<RuntimeSettings>() ?? new RuntimeSettings();
+
+            // Bot
+            services.AddSingleton<IBot, CoreBot>();
+            services.AddOptions()
+                .Configure<CoreBotOptions>(o =>
+                {
+                    o.DefaultLocale = defaultLocale;
+                    o.RootDialog = rootDialog;
+                });
+
+            services.AddSingleton(configuration);
```

Comment:
This is redundant, IConfiguration is already registered by the aspnet core infrastructure.